src/main/java/net/minecraftforge/client/gui/config/ControlCreator.java

```diff
@@ -0,0 +1,519 @@
+        }
+
+        public void addValidator(Predicate<T> newValidator) {
+            validator = validator.and(newValidator);
+        }
+
+        public boolean isValid(T newValue) {
+            return validator.test(newValue);
+        }
+
+        public void addSaver(Consumer<T> newSaver) {
+            saver = saver.andThen(newSaver);
+        }
+
+        /**
+         * Saves a valid new value.
+         */
+        public void save(T newValue) {
+            saver.accept(newValue);
+        }
+
+        public void addUpdateResponder(BiConsumer<Boolean, T> newUpdateResponder) {
+            updateResponder = updateResponder.andThen(newUpdateResponder);
+        }
+
+        public void onUpdate(T newValue) {
+            onUpdate(isValid(newValue), newValue);
+        }
+
+        /**
+         * Responds to value changes.
+         * Called to set up the Interactor after initialisation.
+         * Called when the user interacts with the Widget directly and changes the value.
+         * Called when the user clicks the undo/reset buttons.
+         */
+        public void onUpdate(boolean isValid, T newValue) {
+            updateResponder.accept(isValid, newValue);
+        }
+
+        public <D> void addData(DataKey<D> key, @Nullable D value) {
+            extraData.put(key, value);
+        }
+
+        @Nullable
+        public <D> D getData(DataKey<D> key) {
+            return (D) extraData.get(key);
+        }
+
+        /**
+         * For making a system similar to a type safe map and the capabilities system.
+         * Lets us store arbitrary extra data in a strongly typed manner.
+         */
+        static class DataKey<T> {
+        }
+
+    }
+
+    public static ControlCreator getDefaultCreator() {
+        return DefaultHolder.INSTANCE;
+    }
+
+    public <T> void createAndInitialiseInteractionWidget(Interactor<T> interactor) {
+        createInteractionWidget(interactor);
+        interactor.onUpdate(interactor.initialValue);
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public <T> void createInteractionWidget(Interactor<T> interactor) {
+        T value = interactor.initialValue;
+        if (value instanceof Boolean)
+            createBooleanInteractionWidget((Interactor<Boolean>) interactor);
+        else if (value instanceof Number)
+            createNumericInteractionWidget((Interactor<? extends Number>) interactor);
+        else if (value instanceof Enum<?>)
+            createEnumInteractionWidget((Interactor<Enum>) interactor);
+        else if (value instanceof String)
+            createStringInteractionWidget((Interactor<String>) interactor);
+        else if (value instanceof List<?>)
+            createListInteractionWidget((Interactor<List<?>>) interactor);
+        else
+            throw new IllegalStateException("Don't know how to make a widget for " + value);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> T[] createHolderForUseInLambda(T value) {
```

Comment:
We typically use AtomicWhatever or MutableWhatever (from apache common langs) for use in lambdas

Reply:
`MutableWhatever` seems like that class I couldn't find. I'll fix this up, ty!